extensions/VisualEditor/lib/ve/src/ui/ve.ui.WindowManager.js

```diff
@@ -47,3 +47,41 @@ ve.ui.WindowManager.prototype.getReadyDelay = function () {
 	// HACK: Really this should be measured by OOjs UI so it can vary by theme
 	return 250;
 };
+
+/**
+ * Set dialog size.
+ *
+ * Fullscreen mode will be used if the dialog is too wide to fit in the screen.
+ *
+ * @chainable
+ */
+ve.ui.WindowManager.prototype.updateWindowSize = function ( win ) {
+	// Bypass for non-current, and thus invisible, windows
+	if ( win !== this.currentWindow ) {
+		return;
+	}
+
+	var viewport = OO.ui.Element.static.getDimensions( win.getElementWindow() ),
+		sizes = this.constructor.static.sizes,
+		size = win.getSize();
+
+	if ( size === 'dynamic' && typeof win.getDynamicSize === 'function' ) {
```

Comment:
Rather than duplicating the whole thing here, I think this change of adding a way to set a dynamic size is generic enough to reasonably go into OOjs UI.